src/pages/sa-clients/forms/PriceForm.js

```diff
@@ -0,0 +1,153 @@
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useForm } from 'src/hooks/form'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { DataSets } from 'src/resources/DataSets'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+
+export default function PriceForm({ labels, maxAccess, obj, window, recordId, fetchGridData }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    initialValues: {
+      clientId: recordId,
+      categoryId: null,
+      currencyId: null,
+      priceType: '',
+      value: ''
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      categoryId: yup.string().required(),
+      currencyId: yup.string().required(),
+      value: yup.number().required(),
+      priceType: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: SaleRepository.Price.set,
+        record: JSON.stringify(obj)
+      })
+
+      toast.success(platformLabels.Updated)
+      window.close()
+      fetchGridData()
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      if (obj.clientId) {
+        const res = await getRequest({
+          extension: SaleRepository.Price.get,
+          parameters: `_clientId=${obj.clientId}&_categoryId=${obj.categoryId}&_currencyId=${obj.currencyId}&_priceType=${obj.priceType}`
+        })
+
+        formik.setValues(res.record)
+      }
+    })()
+  }, [])
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.Client}
+      form={formik}
+      maxAccess={maxAccess}
+      editMode={editMode}
+      infoVisible={false}
+      isCleared={false}
+    >
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={InventoryRepository.Category.qry}
+                parameters='_pagesize=30&_startAt=0&_name='
+                name='categoryId'
+                label={labels.category}
+                valueField='recordId'
+                displayField={'name'}
+                displayFieldWidth={1}
+                required
+                values={formik?.values}
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('categoryId', newValue?.recordId)
```

Comment:
![image](https://github.com/user-attachments/assets/56ddc154-e354-41bb-a682-cb2d953f7287)
add || null or an onClear function, and make sure that all the fields used in this task have a workable on clear function